### ISSUE TITLE
is support for gatsby possible?

### ISSUE THREAD

**Issue by hayley** [cj/opal-webpack#36]
So I'll preface all of this with the fact that I'm basically completely new to webpack and js toolsets (my background is middleman/sprockets), but I'm wondering if it would be possible to for opal-webpack to work with:
https://github.com/gatsbyjs/gatsby

I've got the repository with the non-working code here:
https://github.com/hayley/hayley.io

If I run just `gatsby develop`, mysteriously, Opal itself will load properly but I can't call 'reactrb'. If I do `bundle exec gatsby develop`, then I get the following error:

ERROR in ./~/opal-webpack?file=opal%2Fbase&cacheIdentifier=%7B%22opal-loader%22%3A%221.0.9%22%2C%22opal-compiler%22%3A%220.9.4%22%7D&requirable=true!/Users/hayley/.gem/ruby/2.3.1/gems/opal-0.9.4/opal/opal/base.rb
Module not found: Error: Cannot resolve module 'babel' in /Users/hayley/.gem/ruby/2.3.1/gems/opal-0.9.4/opal/opal
 @ ./~/opal-webpack?file=opal%2Fbase&cacheIdentifier=%7B%22opal-loader%22%3A%221.0.9%22%2C%22opal-compiler%22%3A%220.9.4%22%7D&requirable=true!/Users/hayley/.gem/ruby/2.3.1/gems/opal-0.9.4/opal/opal/base.rb 1:21-101

**Comment by wied03** [comment]
#7 is sort of related. It looks like you're doing a `require 'babel'` from Ruby/Opal from the looks of that error. Not sure why you'd do that but if you did, you would want to manually add the Babel node module directory to the load path (see load path section of README). That's what #7 would make easier. That said, don't get why you'd need to require babel from Opal. It's almost as if you're using the React-rails package or something and you have more files going through Opal than you need to.

**Comment by hayley** [comment]
Yeah, it's got me stumped because I'm definitely not doing that. And it only happens when I do a `bundle exec`. My total guess is that bundler is somehow messing up the paths that gatsby is using to build its own things. So then gatsby's build process is looking for vanilla JS requirements in my gem directories.

**Comment by hayley** [comment]
So I'm trying to work around needing bundle exec.

I've read https://github.com/cj/opal-webpack/tree/master/examples/complex and it looks like the following is theoretically what I need to do right:
```
require 'opal'
Opal.use_gem 'reactrb'
```

I'm getting `Uncaught use_gem: undefined method `use_gem' for undefined` though.

**Comment by wied03** [comment]
The examples are really out of date and this has almost been rewritten since then.

**Comment by wied03** [comment]
Are you using react-rails ?

**Comment by hayley** [comment]
No to react-rails.

If it helps, opal files themselves seem to compile just fine. It's when I go to require something (e.g. `require 'opal'` *and* use bundle exec that I have problems.

I got things sort of working by calling reactrb-express (see https://github.com/hayley/hayley.io/blob/master/html.js), which bundles the following into one .js file:
```
gem 'rake'
gem 'opal'
gem 'opal-browser'
gem 'reactrb'
gem 'opal-jquery'
```

However, I'm assuming that in order to actually take advantage of *gatsby* being able to use rb files directly, I'm going to have to figure out how to get everything working through webpack.

Oh, not sure if related, but I also tried doing a `require` from one .rb file to another in the same directory and couldn't get it to work. I tried require_relative.

**Comment by wied03** [comment]
@hayley - Sorry. I've been slammed with other things and just haven't been able to spend time on this project lately.

**Comment by hayley** [comment]
Hey, no worries.

FYI, since @catmando mentioned opening stackoverflow questions for anything reactrb related, I opened http://stackoverflow.com/questions/38549955/is-it-possible-to-use-the-gatsby-static-site-generator-with-reactrb 

I wish I knew more about webpack and such so I could directly help, but let me know if there's any info/help I can provide.
